File: Arrival of the General/arrival_of_the_general.cpp
#include <iostream>
#define ll long long
using namespace std;

int Arrival_of_the_General()
{   int soldiers_count;
    cin>>soldiers_count;
    int height_of_soldier[soldiers_count];
    
    for(int i = 0 ;i < soldiers_count ;i++)
    {cin>>height_of_soldier[i];}
    
    int max_height = height_of_soldier[0];
    int max_height_index = 0;
    int min_height = height_of_soldier[0];
    int min_height_index = 0;
    
    for(int soldier = 1 ;soldier < soldiers_count ;soldier++)
    {
        if(height_of_soldier[soldier] > max_height)
        {   max_height = height_of_soldier[soldier];
            max_height_index = soldier;
        }
        
        if(height_of_soldier[soldier] <= min_height )
        {
            min_height = height_of_soldier[soldier];
            min_height_index = soldier;
        }
    }
    
    
    if(max_height_index < min_height_index)
    return(max_height_index + (soldiers_count - 1 - min_height_index));
    
    return (max_height_index + (soldiers_count - 1 - min_height_index ) - 1);
    
    
}


int main() {
	int remained_test_cases = 1 ;
// 	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   cout<<Arrival_of_the_General();
	    remained_test_cases--;
	}
	return 0;
}

File: Arrival of the General/arrival_of_the_general.py
def Arrival_of_the_General():
    soldiers_count = int(input())
    height_of_soldier = list(map(int,input().split()))
    max_height = height_of_soldier[0]
    max_at_index = 0
    min_height = height_of_soldier[0]
    min_at_index = 0
    
    for i in range(1,soldiers_count):
        if height_of_soldier[i] > max_height:
            max_height = height_of_soldier[i]
            max_at_index = i 
            
        if height_of_soldier[i] <= min_height:
            min_height = height_of_soldier[i]
            min_at_index = i 
    
    if max_at_index < min_at_index:
        return max_at_index + (soldiers_count - 1 - min_at_index)
        
    return max_at_index + (soldiers_count - 1 - min_at_index) - 1     
    
    

remained_test_cases = 1 
# remained_test_cases = int(input())
while remained_test_cases > 0:
    print(Arrival_of_the_General())
    remained_test_cases -= 1 